cmd/podman/common/specgen.go

```diff
@@ -771,3 +775,63 @@ func parseThrottleIOPsDevices(iopsDevices []string) (map[string]specs.LinuxThrot
 	}
 	return td, nil
 }
+
+func parseSecrets(secrets []string) ([]string, map[string]string, error) {
+	secretParseError := errors.New("error parsing secret")
+	var mount []string
+	envs := make(map[string]string)
+	for _, val := range secrets {
+		source := ""
+		secretType := "mount"
+		target := ""
+		envOnly := false
+		split := strings.Split(val, ",")
+
+		// --secret mysecret
+		if len(split) == 1 {
+			source = val
+			mount = append(mount, source)
+			continue
+		}
+		// --secret mysecret,opt=opt
+		maybeSource := strings.SplitN(split[0], "=", 2)
+		if len(maybeSource) == 1 {
+			source = split[0]
+			split = split[1:]
+		}
+		// TODO: implement other secret options
+		for _, val := range split {
+			kv := strings.SplitN(val, "=", 2)
+			switch kv[0] {
+			case "source":
+				source = kv[1]
+			case "type":
+				if kv[1] == "mount" || kv[1] == "env" {
```

Comment:
If it does not match one of these, you should throw an error.

Also if secretType != "" then you should throw an error.